app/controllers/nonsipp/totalvaluequotedshares/QuotedSharesManagedFundsHeldController.scala

```diff
@@ -73,7 +73,9 @@ class QuotedSharesManagedFundsHeldController @Inject()(
                   updatedUserAnswers.set(TotalValueQuotedSharesPage(srn), Money(0.00))
                 }
               } else {
-                baseUpdate
+                baseUpdate.flatMap { updatedUserAnswers =>
+                  updatedUserAnswers.remove(TotalValueQuotedSharesPage(srn))
```

Comment:
I think this removes the value unnecessarily in some scenarios. E.g. when a value greater than 0 is fetched from ETMP. I'll add more details to slack.